specification/app/resource-manager/Microsoft.App/stable/2025-07-01/CommonDefinitions.json

```diff
@@ -0,0 +1,1555 @@
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DaprComponent"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "DaprComponent": {
+      "description": "Dapr Component.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Dapr Component resource specific properties",
+          "type": "object",
+          "properties": {
+            "componentType": {
+              "description": "Component type",
+              "type": "string"
+            },
+            "version": {
+              "description": "Component version",
+              "type": "string"
+            },
+            "ignoreErrors": {
+              "description": "Boolean describing if the component errors are ignores",
+              "type": "boolean",
+              "default": false
+            },
+            "initTimeout": {
+              "description": "Initialization timeout",
+              "type": "string"
+            },
+            "secrets": {
+              "description": "Collection of secrets used by a Dapr component",
+              "type": "array",
+              "items": {
+                "$ref": "./CommonDefinitions.json#/definitions/Secret"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "secretStoreComponent": {
+              "description": "Name of a Dapr component to retrieve component secrets from",
+              "type": "string"
+            },
+            "metadata": {
+              "description": "Component metadata",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/DaprMetadata"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "scopes": {
+              "description": "Names of container apps that can use this Dapr component",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "provisioningState": {
+              "description": "Provisioning state of the Dapr Component.",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "Canceled",
+                "InProgress",
+                "Deleting"
+              ],
+              "type": "string",
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "DaprComponentProvisioningState",
+                "modelAsString": true
+              }
+            },
+            "deploymentErrors": {
```

Comment:
>deploymentErrors [](http://example.com/codeflow?start=13&length=16)

If this is a collection, it should be modeled as an array. If it's not a collection, it should not have a plural name and description.

Reply:
It's stable release api version since [specification/app/resource-manager/Microsoft.App/stable/2022-10-01/CommonDefinitions.json](https://github.com/Azure/azure-rest-api-specs/blob/c2f167ec164a2349235d8aee1b8189b883ec93c6/specification/app/resource-manager/Microsoft.App/stable/2022-10-01/CommonDefinitions.json#L842)

We don't to want to introduce breaking change

Reply:
It's different API versions, under folder name 2025-01-01 or 2022-10-01, each folder for each version as Swagger required.

Reply:
https://github.com/Azure/azure-rest-api-specs/pull/34983/files#diff-b6a4ba0ae617ab6d8125ed7992b46d19a45542b7a0dd8363347a24f8bf2ea5b6.
deploymentErrors  got arm approval by [rkmanda](https://github.com/rkmanda)